### ISSUE TITLE
New Rule: secretlint-rule-secret-patterns

### ISSUE THREAD

**Issue by azu** [secretlint/secretlint#130]
It is generic rule (not recommended).

Thie rule just use [@textlint/regexp-string-matcher](https://github.com/textlint/regexp-string-matcher#regexp-like-string) for matching content with rule's `options`.

```json5
{
  "rules": [
    {
      "id": "@secretlint/ecretlint-rule-secret-patterns",
      "options": {
        "pattens": [
          "/IT IS SECRET/i",
          "/API_TOKEN=\w{40}/i"
        ],
        "patternFiles": ["./secret.yml"] // need?
      }
    }
  ]
}
```

This rule allow to define secret patterns by user.
In other words, it is just user defined NG words list.

It similar concept with [textlint-rule-prh](https://github.com/textlint-rule/textlint-rule-prh).
This rule allow a project to have own dictionary for secretlint.

## TODO

- [ ] Require suitable rule name
- [ ] Implement

**Comment by PseudoCoding** [comment]
Thoughts on a rule like this:

```json
{
  "rules": [
    {
      "id": "@secretlint/secretlint-rule-pattern",
      "options": {
        "patterns": [
          {
            "name": "password=",
            "pattern": "/password\\s*=\\s*(?<password>[\\w\\d!@#$%^&(){}\\[\\]:\";'<>,.?\/~`_+-=|]{1,256})\\b.*\/gi"
          },
          {
            "name": "client_secret=",
            "pattern": "/client_secret\\s*=\\s*(?<password>[\\w\\d!@#$%^&(){}\\[\\]:\";'<>,.?\/~`_+-=|]{1,256})\\b.*\/gi"
          }
        ]
      }
    }
  ]
}
```

where the message would look like this:

```js
export const messages = {
    PATTERN: {
        en: (props: { PATTERN_NAME: string, CREDENTIAL: string }) => `found pattern matching ${props.PATTERN_NAME}: ${props.CREDENTIAL}`
    },
};
```

**Comment by azu** [comment]
I like it.

**Comment by PseudoCoding** [comment]
I'll try to get a PR open soon then!

**Comment by azu** [comment]
resolved by #139 Thanks @PseudoCoding

**Comment by PseudoCoding** [comment]
no problem! how do we get the package published to NPM?

**Comment by PseudoCoding** [comment]
ah nevermind. I see #140

**Comment by azu** [comment]
Implemted #140 

Thanks @PseudoCoding 🎉
